dpdispatcher/dp_cloud_server_context.py

```diff
@@ -118,11 +144,21 @@ def download(self, submission):
                 if 'result_url' in each and each['result_url'] != '' and each['status'] == 2:
                     job_hash = job_hashs[each['task_id']]
                     job_infos[job_hash] = each
-        for hash, info in job_infos.items():
-            result_filename = hash + '_back.zip'
+        for job_hash, info in job_infos.items():
+            result_filename = job_hash + '_back.zip'
             target_result_zip = os.path.join(self.local_root, result_filename)
             self.api.download_from_url(info['result_url'], target_result_zip)
             zip_file.unzip_file(target_result_zip, out_dir=self.local_root)
+            try:
+                if self.remote_profile.get('keep_backup', True):
+                    # move to backup directory
+                    os.makedirs(os.path.join(self.local_root, 'backup'), exist_ok=True)
+                    shutil.move(target_result_zip,
+                                os.path.join(self.local_root, 'backup', os.path.split(target_result_zip)[1]))
+                else:
+                    os.remove(target_result_zip)
+            except Exception as e:
```

Comment:
Use OSError, shutil.Error instead of Exception.

By the way, the code is duplicated here?

Reply:
fixed